elide-graphql/src/main/java/com/yahoo/elide/graphql/GraphQLConversionUtils.java

```diff
@@ -133,19 +135,31 @@ public GraphQLEnumType classToEnumType(Type<?> enumClazz) {
         if (enumConversions.containsKey(enumClazz)) {
             return enumConversions.get(enumClazz);
         }
+        GraphQLEnumType enumResult = buildClassToEnumType(enumClazz, Function.identity(), (value) -> true);
+        enumConversions.put(enumClazz, enumResult);
+        return enumResult;
+    }
 
-        Enum [] values = (Enum []) enumClazz.getEnumConstants();
+    /**
+     * Converts an enum to a GraphQLEnumType.
+     * @param enumClazz the Enum to convert
+     * @param nameProcessor the Enum to convert
+     * @param enumClazz the Enum to convert
+     * @return A GraphQLEnum type for class.
+     */
+    public GraphQLEnumType buildClassToEnumType(Type<?> enumClazz, Function<String, String> nameProcessor,
+            Predicate<Enum<?>> filter) {
+        Enum<?> [] values = (Enum []) enumClazz.getEnumConstants();
 
-        GraphQLEnumType.Builder builder = newEnum().name(nameUtils.toOutputTypeName(enumClazz));
+        GraphQLEnumType.Builder builder = newEnum().name(nameProcessor.apply(nameUtils.toOutputTypeName(enumClazz)));
 
-        for (Enum value : values) {
-            builder.value(value.toString(), value);
+        for (Enum<?> value : values) {
+            if (filter.test(value)) {
+                builder.value(value.toString(), value);
+            }
         }
 
         GraphQLEnumType enumResult = builder.build();
-
-        enumConversions.put(enumClazz, enumResult);
```

Comment:
Why did we remove this line?

Reply:
Agree this method is a bit confusing. From what I gather it's not possible to set a graphql enum and indicate that only a subset of values are acceptable. Instead I needed to create a graphql enum per entity. eg. instead of just ElideRelationshipOp I needed to create one per entity, eg. ElideRelationshipOpDownloads, ElideRelationshipOpGroup, ElideRelationshipOpProduct and then indicate the acceptable values for each. In this case the enums needed to be deduplicated based on the graphql enum name and not the enum type. I have changed this to create a specific method for this use case instead of trying to modify the existing method to cater to this.